src/oauth2.js

```diff
@@ -2,10 +2,24 @@ import querystring from 'query-string'
 import cuid from 'cuid'
 import openPopup from './util/popup'
 
-const listenForCredentials = (popup, state, resolve, reject) => {
+const listenForCredentials = (popup, state, resolve, reject, config) => {
   let hash
+  let credentialsFound = true
   try {
     hash = popup.location.hash
+    if (config.parseQuery) {
+      credentialsFound = false
+      const parsedQuery = querystring.parse(popup.location.search)
+      if (parsedQuery[config.accessTokenKey]) {
+        hash = hash ? hash + '&access_token=' + parsedQuery[config.accessTokenKey] : '#access_token=' + parsedQuery[config.accessTokenKey]
```

Comment:
I don't understand why you are adding this to the hash. Why not get the response from the hash or the query:
```
if (config.parseQuery) {
  // get from query
} else {
  // get from hash
}
```

Reply:
This is because of a strange case with IG's implicit oauth mechanism. See, with IG the response comes back partially in both query params and hash params. So, I still need to parse the hash params and the query params at the same time. In the above code, I have tried to look for everything of interest that can be found in query and hash params and concat them together into the hash variable so that it can be used later on.